lib/team_stats.rb

```diff
@@ -0,0 +1,208 @@
+module TeamStats
+
+  def team_info(team_id)
+    info = {}
+    info["team_id"] = @teams[team_id].team_id
+    info["franchiseId"] = @teams[team_id].franchiseId
+    info["teamName"] = @teams[team_id].teamName
+    info["abbreviation"] = @teams[team_id].abbreviation
+    info["link"] = @teams[team_id].link
+    info
+  end
+
+
+  def best_season(team_id)
+    worst_and_best_season(team_id)[1]
+  end
+
+  def worst_season(team_id)
+    worst_and_best_season(team_id)[0]
+  end
+
+  def average_win_percentage(team_id)
+    (calculate_percents[2][team_id]/100).round(2)
+  end
+
+  def most_goals_scored(team_id)
+    @game_teams[team_id].max_by{ |object| object.goals }.goals
+  end
+
+  def fewest_goals_scored(team_id)
+    @game_teams[team_id].min_by{ |object| object.goals }.goals
+  end
+
+  def favorite_opponent(team_id)
+    generate_percent(team_id)[0]
+  end
+
+  def rival(team_id)
+    generate_percent(team_id)[1]
+  end
+
+
+  def biggest_team_blowout(team_id)
+    generate_goals_difference(team_id)[0]
+  end
+
+  def worst_loss(team_id)
+    generate_goals_difference(team_id)[1].abs
+  end
+
+  def head_to_head(team_id)
+    opponents_and_win_percentage = {}
+    generate_percent(team_id)[3].each do |opponent, percent|
+      opponents_and_win_percentage[@teams[opponent].teamName] = (percent/100).round(2)
+    end
+    opponents_and_win_percentage
+  end
+
+  def seasonal_summary(team_id)
+    summary = {}
+    @game_teams[team_id].each do |game|
+      summary[@games[game.game_id].season]
+      summary[@games[game.game_id].season]
+    end
+  end
+
+  def generate_post_and_regular(team_id)
+    game_ids = []
+    post_and_reg = { }
+    @game_teams[team_id].each {|game| game_ids << game.game_id}
+    game_ids.each do |id|
+      # require "pry"; binding.pry
+      if @games[id].type == "Postseason" && !post_and_reg.has_key?(@games[id].season)
+        post_and_reg[@games[id].season] = {}
+        post_and_reg[@games[id].season][:post_season] = [id]
+      elsif @games[id].type == "Postseason" && post_and_reg.has_key?(@games[id].season)
+          post_and_reg[@games[id].season] = {}
+          post_and_reg[@games[id].season][:post_season] = []
+          post_and_reg[@games[id].season][:post_season] << id
+      elsif @games[id].type == "Regular Season" && !post_and_reg.has_key?(@games[id].season)
+        post_and_reg[@games[id].season] = {}
+        post_and_reg[@games[id].season][:regular_season] = [id]
+      elsif @games[id].type == "Regular Season" && post_and_reg.has_key?(@games[id].season)
+        post_and_reg[@games[id].season] = {}
+        post_and_reg[@games[id].season][:regular_season] = []
+        post_and_reg[@games[id].season][:regular_season] << id
+      end
+    end
+    post_and_reg
+  end
+
```

Comment:
do we need to write the formula for this method? or delete it?

Reply:
we will be using it, it is in progress